test/wdio.conf.ts

```diff
@@ -307,8 +309,18 @@ export const config: Options.Testrunner = {
    * @param {Boolean} result.passed    true if test has passed, otherwise false
    * @param {Object}  result.retries   informations to spec related retries, e.g. `{ attempts: 0, limit: 0 }`
    */
-  // afterTest: function(test, context, { error, result, duration, passed, retries }) {
-  // },
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  afterTest: async (test, __, { passed }) => {
+    if (!passed) {
+      const screenshotDir = join(__dirname, 'screenshots');
+      console.log(`Test run failed. Saving a screenshot of the failure here: ${screenshotDir}`);
+      if (!existsSync(screenshotDir)) {
+        mkdirSync(screenshotDir, { recursive: true });
+      }
+      const sanitizedTestTitle = test.title.replace(/[^a-zA-Z0-9 ]/g, "");
+      await browser.saveScreenshot(join(screenshotDir, `${test.parent} - ${sanitizedTestTitle}.png`))
+    }
+  },
```

Comment:
@randi274 I think this would be cool if it was a utility function, and tests could (also) call it when we need to debug something in GHA.  I'm thinking:
1. create a new file, `/test/utilities/screenshot.ts`
2. within the file, add:
```
export async function saveScreenshot(parentPath: string, testTitle: string): Promise<void> {
  const screenshotDir = join(__dirname, 'screenshots');
  console.log(`Test run failed. Saving a screenshot of the failure here: ${screenshotDir}`);
  if (!existsSync(screenshotDir)) {
    mkdirSync(screenshotDir, { recursive: true });
  }
  const sanitizedTestTitle = testTitle.replace(/[^a-zA-Z0-9 ]/g, "");
  const screenshotPath = join(screenshotDir, `${parentPath} - ${sanitizedTestTitle}.png`);
  await browser.saveScreenshot(screenshotPath);
}
```
(warning-this is untested code)
3. and then in wdio.conf.ts:
```
  // eslint-disable-next-line @typescript-eslint/no-misused-promises
  afterTest: async (test, __, { passed }) => {
    if (passed) {
      return;
    }    

    await utilities.saveScreenshot(test.parent, test.title);
  }
```

Reply:
Nice! I think this would fit more with the conventions for the rest of the repo, so I'll try moving this one out as well.

Building on this, I could also distinguish between a screenshot for a test failure and a screenshot we want to save to make debugging easier.

Reply:
Also interesting thing I learned - breakpoints work in the utility function called from the afterTest hook, but they *don't* work in the wdio.conf.ts